TLM/TLM/Traffic/Data/SpeedLimit.cs

```diff
@@ -0,0 +1,176 @@
+﻿using System;
+using System.Collections.Generic;
+using TrafficManager.State;
+using TrafficManager.UI;
+using UnityEngine;
+
+namespace TrafficManager.Traffic.Data {
+	public enum SpeedUnit {
+		CurrentlyConfigured,	// Currently selected in the options menu
+		Kmph,
+		Mph
+	}
+
+	public enum MphSignStyle {
+		SquareUS = 0,
+		RoundUK = 1,
+		RoundGerman = 2,
+	}
+
+	/// <summary>
+	/// Defines a speed limit value with default Kmph and display value of Mph
+	/// for when the option is set to display Mph. The engine still uses kmph.
+	/// </summary>
+	public struct SpeedLimit {
+		public const float SPEED_TO_KMPH = 50.0f; // 1.0f equals 50 km/h
+		private const ushort LOWER_KMPH = 10;
+		public const ushort UPPER_KMPH = 140;
+		private const ushort KMPH_STEP = 10;
+		public const int BREAK_PALETTE_COLUMN_KMPH = 8; // palette shows N in a row, then break and another row
+
+		private const float SPEED_TO_MPH = 32.06f; // 50 km/h converted to mph
+		private const ushort LOWER_MPH = 5;
+		public const ushort UPPER_MPH = 90;
+		private const ushort MPH_STEP = 5;
+		public const int BREAK_PALETTE_COLUMN_MPH = 10; // palette shows M in a row, then break and another row
+
+		private const float LOWER_SPEED = 0.1f; 
+		private const float UPPER_SPEED = 2 * 10.0f; // 1000 km/h
+
+		/// <summary>
+		/// Produces list of speed limits to offer user in the palette
+		/// </summary>
+		/// <param name="unit">What kind of speed limit list is required</param>
+		/// <returns>List from smallest to largest speed with the given unit. Zero (no limit) is not added to the list.
+		/// The values are in-game speeds as float.</returns>
+		public static List<float> EnumerateSpeedLimits(SpeedUnit unit) {
+			var result = new List<float>();
+			switch (unit) {
+				case SpeedUnit.Kmph:
+					for (var km = LOWER_KMPH; km <= UPPER_KMPH; km += KMPH_STEP) {
+						result.Add(km / SPEED_TO_KMPH);
+					}
+					break;
+				case SpeedUnit.Mph:
+					for (var mi = LOWER_MPH; mi <= UPPER_MPH; mi += MPH_STEP) {
+						result.Add(mi / SPEED_TO_MPH);
+					}
+					break;
+				case SpeedUnit.CurrentlyConfigured:
+					// Automatically choose from the config
+					return GlobalConfig.Instance.Main.DisplaySpeedLimitsMph
+						       ? EnumerateSpeedLimits(SpeedUnit.Mph)
+						       : EnumerateSpeedLimits(SpeedUnit.Kmph);
+			}
+
+			return result;
+		}
+
+		public static string ToMphPreciseString(float speed) {
+			if (IsZero(speed)) {
+				return Translation.GetString("Speed_limit_unlimited");
+			}
+
+			return ToMphPrecise(speed) + " mph";
```

Comment:
Can we make this capitalized to match the lang file?
```suggestion
			return ToMphPrecise(speed) + "MPH";
```

Reply:
I am happy to make it uppercase, but not happy with the fact that the space is gone.

Reply:
Fixed, sorry that was an accident.